quickwit-cli/src/split.rs

```diff
@@ -291,6 +336,25 @@ async fn list_split_cli(args: ListSplitArgs) -> anyhow::Result<()> {
     Ok(())
 }
 
+async fn delete_splits_cli(args: DeleteSplitsArgs) -> anyhow::Result<()> {
+    debug!(args = ?args, "delete-splits");
+
+    let quickwit_config = load_quickwit_config(&args.config_uri, None).await?;
+    let metastore_uri_resolver = quickwit_metastore_uri_resolver();
+    let metastore = metastore_uri_resolver
+        .resolve(&quickwit_config.metastore_uri())
+        .await?;
+    let split_ids: Vec<&str> = args
+        .split_ids
+        .iter()
+        .map(|split_id| split_id.as_ref())
+        .collect();
+    metastore
+        .mark_splits_for_deletion(&args.index_id, &split_ids)
+        .await?;
```

Comment:
Note that we don't run GC here, so "delete" here is misleading. Some options:
- rename the option `--mark-as-deleted`
- run the GC
- run the GC and add a `--no-gc` option
- your awesome idea here :)

Let me know what you think and I'll make the appropriate changes.

Reply:
I'm going to rename the command `mark-for-deletion` and provide a `mark` alias. The gc will delete the splits later. One can always run the gc manually with `quickwit index gc`  and a short grace period if necessary.